### PR TITLE
Chore: use fork-ts-checker plugin for eslint check

### PR DESCRIPTION
**Pull request by zoltanbedi**
**What this PR does / why we need it**:

This pr removes eslint-loader from webpack.configs so the build time can be faster. Also upgrades fork-ts-checker-plugin to version 5 so it will be compatible with webpack 5.

Some measurements on my pc:

yarn start (without eslint-loader): 73253ms
yarn start (with eslint-loader): 113371ms

yarn build (without eslint-loader): 248.47s
yarn build (with eslint-loader): 289.10s

in both scripts we can compile almost **~1 minute** faster this way.

### CODE REVIEW

**Review by aknuds1 — APPROVED** [review]
Not really familiar with it, but LGTM

**Review by torkelo — APPROVED** [review]
Looks good!

**Review by stevenvachon — APPROVED** [review]
Nice! I wasn't aware of this and it looks as though we may have been doubling up checks. Perhaps we should also change our "lint" npm scripts to use webpack, so that our paths are only in one place.

**Comment by zoltanbedi** [comment]
> Perhaps we should also change our "lint" npm scripts to use webpack, so that our paths are only in one place.

I'd rather not do that. This change only going to lint packages that is compiled through webpack. So no test files for example.

I had to do some changes because for my first try it only checked files under public/app. So I've added package paths that are compiled. Compilation was still fast but eslint check become slower. So I've enabled cache for eslint which made it **BLAZING** fast again.

File: scripts/webpack/webpack.dev.js [hunk 1]
```diff
@@ -89,7 +89,17 @@ module.exports = (env = {}) =>
...
             eslint: {
               enabled: true,
-              files: ['public/app/**/*.{ts,tsx}', 'e2e/suite1/**/*.{ts,tsx}', 'public/test/**/*.{ts,tsx}'],
+              files: [
+                'public/app/**/*.{ts,tsx}',
+                'packages/grafana-ui/src/**/*.{ts,tsx}',
+                'packages/grafana-data/src/**/*.{ts,tsx}',
+                'packages/grafana-runtime/src/**/*.{ts,tsx}',
+                'packages/grafana-e2e-selectors/src/**/*.{ts,tsx}',
+                'packages/jaeger-ui-components/src/**/*.{ts,tsx}',
```
Comment (stevenvachon) [inline scripts/webpack/webpack.dev.js:98]: wouldn't `packages/**/*.{ts,tsx}` suffice?

Comment (zoltanbedi) [reply scripts/webpack/webpack.dev.js:98]: good catch but sadly I couldn’t make that work. It failed with an error I couldn’t manage to solve.

Comment (stevenvachon) [reply scripts/webpack/webpack.dev.js:98]: Okay, maybe add a comment explaining why. Aside from that, the changes LGTM.

**Review by stevenvachon — APPROVED** [review]

**Pull request merged**